### PR TITLE
Update Dockerfile to add Catalyst

### PR DESCRIPTION
**Pull request by Scitator**
Hi, just want to add [Catalyst](https://github.com/catalyst-team/catalyst) - high-level PyTorch framework for reproducible and fast R&D.

### CODE REVIEW

**Comment by rosbo** [comment]
Hi @Scitator,

Can you add tests to prevent regression. Examples can be found under the `/tests` folder in this repo.

Thank you

**Comment by Scitator** [comment]
Hi @rosbo 

So minimal test case is done ;)
All other test are done during catalyst CI.

**Comment by rosbo** [comment]
Hi @Scitator,

If you can add a simple smoke test that exercise a common path of catalyst that would be great.

Because we are installing 1000s of packages in this image, some dependencies may get reinstalled to a different version which may cause you package to fail inside the Kaggle image even though it works in your own CI (given the dependency may differ).

Thank you

**Comment by Scitator** [comment]
@rosbo Would mnist test be enough?

File: tests/test_catalyst.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        data_transform = transforms.Compose([
+            transforms.ToTensor(),
+            transforms.Normalize((0.1307, ), (0.3081, ))
+        ])
+
+        loaders = collections.OrderedDict()
+
+        trainset = torchvision.datasets.MNIST(
+            root='./data', train=False,
+            download=True, transform=data_transform)
```
Comment (rosbo) [inline tests/test_catalyst.py]: The tests don't have internet access (they are hermetic), but you can use `np.random` to generate a train set:
https://github.com/Kaggle/docker-python/blob/master/tests/test_keras.py#L34

**Comment by Scitator** [comment]
@rosbo Could you please check example now?

**Comment by Scitator** [comment]
Hmm...why do you think tests have failed?

File: tests/test_catalyst.py (hunk 1 above)
Comment (Scitator) [reply tests/test_catalyst.py]: @rosbo have we resolve the conversation?

**Comment by rosbo** [comment]
@Scitator 

```
======================================================================

ERROR: test_version (test_catalyst.TestCatalyst)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/input/tests/test_catalyst.py", line 85, in test_version

    self.assertIsNotNone(catalyst.__version__)

NameError: name 'catalyst' is not defined
```

**Comment by rosbo** [comment]
It seems like the installation probably failed. I will look at the build logs.

File: tests/test_catalyst.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        # model runner
+        runner = SupervisedRunner()
+
+        # model training
+        runner.train(
+            model=model,
+            criterion=criterion,
+            optimizer=optimizer,
+            loaders=loaders,
+            logdir=logdir,
```
Comment (rosbo) [inline tests/test_catalyst.py]: Do you need to call some sort of `Close` method?

The tests shows a ResourceWarning:

```
unclosed file <_io.TextIOWrapper name='/working/logs/log.txt' mode='a' encoding='UTF-8'>
```

File: tests/test_catalyst.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
+import unittest
+import collections
+import numpy as np
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torchvision
+import torchvision.transforms as transforms
+
```
Comment (rosbo) [inline tests/test_catalyst.py]: You need to add `import catalyst` here to be able to call `catalyst.__version__`

**Comment by Scitator** [comment]
@rosbo Looks like all checks have passed :)
`unclosed file` issue will be resolved in Catalyst 19.10 release

**Pull request merged**